File: mobile/src/release/java/jp/satorufujiwara/kotlin/AppComponent.java
package jp.satorufujiwara.kotlin;

import dagger.Component;
import jp.satorufujiwara.kotlin.data.DataModule;

@AppScope
@Component(
        modules = {
                AppModule.class,
                DataModule.class
        })
public interface AppComponent extends BaseAppComponent {

    void inject(DaggerApp app);

    static class Initializer {

        private Initializer() {
        }

        public static AppComponent init(DaggerApp app) {
            return DaggerAppComponent.builder()
                    .appModule(new AppModule(app))
                    .dataModule(new DataModule())
                    .build();
        }
    }

}

File: mobile/src/release/kotlin/jp/satorufujiwara/kotlin/AppComponent.kt
package jp.satorufujiwara.kotlin

import dagger.Component
import jp.satorufujiwara.kotlin.data.DataModule
import jp.satorufujiwara.kotlin.ui.FluxModule

@AppScope
@Component(modules = arrayOf(AppModule::class, DataModule::class, FluxModule::class))
public interface AppComponent : MainAppComponent {

    fun inject(app: KotlinApp)

    object Initializer {

        fun init(app: KotlinApp): AppComponent =
                DaggerAppComponent.builder()
                        .appModule(AppModule(app))
                        .dataModule(DataModule())
                        .build()
    }

}
